src/scraping/collector.py

```
@@ -0,0 +1,123 @@
+                    if response.status != 200:
+                        break
+                    
+                    data = await response.json()
+                    if not data.get("listings"):
+                        break
+                        
+                    listings.extend(data["listings"])
+                    page += 1
+                    await asyncio.sleep(self.delay)
+                    
+            except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)